packages/lib/CalendarService.ts

```diff
@@ -187,23 +187,48 @@ export default abstract class BaseCalendarService implements Calendar {
           additionalInfo: {},
         };
       }
-
+      let calendarEvent: CalendarEventType;
       const eventsToUpdate = events.filter((e) => e.uid === uid);
       return Promise.all(
-        eventsToUpdate.map((e) => {
+        eventsToUpdate.map((eventItem) => {
+          calendarEvent = eventItem;
           return updateCalendarObject({
             calendarObject: {
-              url: e.url,
-              data: iCalString,
-              etag: e?.etag,
+              url: calendarEvent.url,
+              data: iCalString?.replace(/METHOD:[^\r\n]+\r\n/g, ""),
+              etag: calendarEvent?.etag,
             },
             headers: this.headers,
           });
         })
-      ).then((p) => p.map((r) => r.json() as unknown as NewCalendarEventType));
+      ).then((responses) =>
+        responses.map((response) => {
+          if (response.status > 199 && response.status < 207) {
```

Comment:
```suggestion
          if (response.status >= 200 && response.status < 300) {
```

Everything between 200 and 299 should be considered success.